packages/account/src/Sections/Security/ClosingAccount/closing-account-reason-form.tsx

```diff
@@ -0,0 +1,266 @@
+            .map(val => val[0])
+            .toString();
+        const is_other_trading_platform__has_value = !!values.other_trading_platforms.length;
+        const is_to_do_improve_has_value = !!values.do_to_improve.length;
+        if (is_other_trading_platform__has_value) {
+            selected_reasons = `${selected_reasons}, ${values.other_trading_platforms}`;
+        }
+        if (is_to_do_improve_has_value) {
+            selected_reasons = `${selected_reasons}, ${values.do_to_improve}`;
+        }
+
+        return selected_reasons.replace(/(\r\n|\n|\r)/gm, ' ');
+    };
+
+    const selectedReasonsForCloseAccount = (values: TClosingAccountFormValues) =>
+        Object.entries(values).filter(
+            ([key, value]) => !['other_trading_platforms', 'do_to_improve'].includes(key) && value
+        );
+
+    return (
+        <Formik initialValues={initial_form_values} validate={validateFields} onSubmit={handleSubmitForm}>
+            {({ values, setFieldValue, errors, handleChange, dirty, isValid }) => (
+                <Form>
+                    {getCloseAccountReasonsList().map(reason => (
+                        <Field name={reason.name} key={reason.name}>
+                            {({ field }: FieldProps) => (
+                                <Checkbox
+                                    {...field}
+                                    disabled={
+                                        is_checkbox_disabled && !values[field.name as keyof TClosingAccountFormValues]
+                                    }
+                                    className='closing-account-reasons__checkbox'
+                                    label={reason.label}
+                                    onChange={() => {
+                                        handleChangeCheckbox(values, field.name, setFieldValue);
+                                    }}
+                                />
+                            )}
+                        </Field>
+                    ))}
+                    <Field name='other_trading_platforms'>
+                        {({ field }: FieldProps) => (
+                            <Input
+                                {...field}
+                                className='closing-account-reasons__input'
+                                data-lpignore='true'
+                                autoComplete='off'
+                                type='textarea'
+                                placeholder={localize(
+                                    'If you don’t mind sharing, which other trading platforms do you use?'
+                                )}
+                                name='other_trading_platforms'
+                                value={values.other_trading_platforms}
+                                max_characters={CHARACTER_LIMIT}
+                                onChange={e => handleInputChange(e, values.other_trading_platforms, handleChange)}
```

Comment:
Why do we need our own `onChange` handler?

Reply:
we have a restriction on the total number of characters in the input textbox